webpack/assets/javascripts/react_app/components/hosts/interfaces/InterfaceRow/index.js

```diff
@@ -0,0 +1,164 @@
+import UUID from 'uuid/v1';
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { OverlayTrigger, Tooltip } from 'patternfly-react';
+import $ from 'jquery';
+
+import { translate as __, sprintf } from '../../../../../react_app/common/I18n';
+import { noop } from '../../../../common/helpers';
+import { fqdn } from '../../../../../foreman_hosts';
+
+class InterfaceRow extends React.Component {
+  getHiddenFields() {
+    return document.getElementById(`interfaceHidden${this.props.id}`);
+  }
+
+  isVirtual() {
+    const virtualTypes = ['Nic::Bond', 'Nic::Bridge'];
+    return this.props.virtual || virtualTypes.indexOf(this.props.type) >= 0;
+  }
+
+  nicInfo() {
+    if (this.isVirtual()) {
+      // common virtual
+      if (this.props.attachedTo !== '')
+        return sprintf(__('virtual attached to %s'), this.props.attachedTo);
+      return __('virtual');
+    }
+
+    // provider specific
+    if (typeof window.providerSpecificNICInfo === 'function')
+      return window.providerSpecificNICInfo($(this.getHiddenFields()));
```

Comment:
I agree, this is an anti-pattern and creates technical debt.  I like @amirfefer's suggestion of using redux to communicate between jquery and react.

Reply:
well, I can clearly see that it is an anti-pattern :) I just am not sure what to do about it, I can shove it in the webpack, to look more clean. But the method is in the window, because the technical debt from past. I am all for moving it in the webpack, but it will throw away the support to be extendible by more providers the same way it was - basically it will throw the API support. And I am not sure till what level we can do that :thinking: I was thinking about some deprecations, but I don't know what is the best way to do this thinks in the future.
I will try to talk about it with @amirfefer, but I don't feel like it should be addressed in this PR.